File: ProblemSolve/946.cpp
/****************************
 * Validate Stack Sequences *
 ****************************/

class Solution {
public:
    bool validateStackSequences(vector<int>& pushed, vector<int>& popped) {
        int len = pushed.size();
        stack<int> st;
        
        int popIdx = 0;
        
        for(int p : pushed){
            st.push(p);
            
            while(!st.empty() && st.top() == popped[popIdx]){
                st.pop();
                popIdx++;
            }
        }
                
        return popIdx == len;
    }
};

File: ProblemSolve/946.cs
/****************************
 * Validate Stack Sequences *
 ****************************/

public class Solution {
    public bool ValidateStackSequences(int[] pushed, int[] popped) {
        int pushedSize = pushed.Length;
        
        Stack<int> st = new Stack<int>();
        
        int popIdx = 0;
        
        foreach(int p in pushed){
            st.Push(p);
            
            while(st.Count != 0 && st.Peek() == popped[popIdx]){
                st.Pop();
                popIdx++;
            }
        }
        
        return popIdx == pushedSize;
    }
}